src/L2DebtManager.sol

```diff
@@ -1,262 +1,502 @@
+     */
+    function convertUsdcToCollateralToken(
+        address collateralToken,
+        uint256 debtUsdcAmount
+    ) public view returns (uint256) {
+        if (collateralToken != address(weETH))
+            revert UnsupportedCollateralToken();
+        return (debtUsdcAmount * 1e18) / priceProvider.getWeEthUsdPrice();
+    }
 
-        return (debtValue * 1e4) / collateralValue; // result in basis points
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function convertCollateralTokenToUsdc(
+        address collateralToken,
+        uint256 collateralAmount
+    ) public view returns (uint256) {
+        if (collateralToken != address(weETH))
+            revert UnsupportedCollateralToken();
+        return (collateralAmount * priceProvider.getWeEthUsdPrice()) / 1e18;
     }
 
-    function remainingBorrowingCapacityInUSDC(
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function getCollateralValueInUsdc(
         address user
     ) public view returns (uint256) {
-        uint256 eEthPriceInUSDC = getEEthPriceInUsdc();
-        uint256 collateralValue = (_collaterals[user] * eEthPriceInUSDC) / 1e18;
-        uint256 maxBorrowingAmount = (collateralValue * liquidationThreshold) /
-            1e4;
         return
-            maxBorrowingAmount > _borrowings[user]
-                ? maxBorrowingAmount - _borrowings[user]
-                : 0;
+            (_userCollateral[user][address(weETH)] *
+                priceProvider.getWeEthUsdPrice()) / 1e18;
     }
 
-    function liquidEEthAmount() public view returns (uint256) {
-        return eETH.balanceOf(address(this)) - totalCollateralAmount;
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function collateralTokens()
+        external
+        view
+        returns (address[] memory tokens)
+    {
+        tokens = new address[](1);
+        tokens[0] = address(weETH);
     }
 
-    function liquidUsdcAmount() public view returns (uint256) {
-        return USDC.balanceOf(address(this)) - totalBorrowingAmount;
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function setLiquidationThreshold(
+        uint256 newThreshold
+    ) external onlyRole(ADMIN_ROLE) {
+        emit LiquidationThresholdUpdated(_liquidationThreshold, newThreshold);
+        _liquidationThreshold = newThreshold;
     }
 
-    // returns eETH price in USDC
-    function getEEthPriceInUsdc() private view returns (uint256) {
-        // TODO: replace it with the actual oracle interaction
-        return eEthPriceInUSDC;
+    /**
+     * @inheritdoc IL2DebtManager
+     */
+    function depositCollateral(address token, uint256 amount) external {
+        if (token != address(weETH)) revert UnsupportedCollateralToken();
+
+        IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
+        _totalCollateralAmounts[token] += amount;
+        _userCollateral[msg.sender][token] += amount;
+
+        emit DepositedCollateral(msg.sender, token, amount);
```

Comment:
can we emit some evetns for tracking {_totalCollateralAmounts, _totalBorrowingAmount}

Reply:
Sure will add some events for that.

Reply:
When should I emit these events? I think it would be overwhelming if we add this event to deposit collateral/borrow/repay functions. Should I add a function to get this state: (Collaterals(token, amount), total collateral in usdc, total borrowings)?

Reply:
Something like this?

```
    /**
     * @notice Function to fetch the current state of collaterals and borrowings.
     * @return totalCollaterals Array of collaterals in tuple(address token, uint256 amount) format.
     * @return totalCollateralInUsdc Total collateral value in USDC.
     * @return totalBorrowings Total borrowing value in USDC.
     */
    function getCurrentState()
        external
        view
        returns (
            TokenData[] memory totalCollaterals,
            uint256 totalCollateralInUsdc,
            uint256 totalBorrowings
        );